core/trino-main/src/main/java/io/trino/execution/FutureStateChange.java

```diff
@@ -34,6 +34,7 @@ public class FutureStateChange<T>
     @GuardedBy("listeners")
     private final Set<SettableFuture<T>> listeners = new HashSet<>();
 
+    @SuppressWarnings("CollectionUndefinedEquality")
```

Comment:
What’s this and why does it need to be suppressed? Can we fix the issue instead?

Reply:
No. This warns that Listenable and Settable futures equals method is flawed

Reply:
Explained below.